ygen/gogen.go

```diff
@@ -1973,3 +1982,35 @@ func writeIfNotEmpty(b *bytes.Buffer, s string) {
 		b.WriteString(s)
 	}
 }
+
+// goLeafDefault returns the default value of the leaf e if specified. If it
+// is unspecified, the value specified by the type is returned if it is not nil,
+// otherwise nil is returned to indicate no default was specified.
+func goLeafDefault(e *yang.Entry, t *mappedType) *string {
+	if e.Default != "" {
+		if t.isEnumeratedValue {
+			return enumDefaultValue(t.nativeType, e.Default, goEnumPrefix)
+		}
+		return quoteDefault(&e.Default, t.nativeType)
+	}
+
+	if t.defaultValue != nil {
+		return quoteDefault(t.defaultValue, t.nativeType)
+	}
+
+	return nil
+}
+
+// quoteDefault adds quotation marks to the value string if the goType specified
+// is a string, and hence requires quoting.
+func quoteDefault(value *string, goType string) *string {
+	if value == nil {
+		return nil
+	}
+
+	if goType == "string" {
+		return ygot.String(fmt.Sprintf(`"%s"`, *value))
```

Comment:
(optional) We could replace this with %q:

return ygot.String(fmt.Sprintf("%q", *value))

Reply:
Oh, neat! I didn't know about %q. Fixed.